### DISCUSSION TITLE
Where can I find the constructor of System.Numerics.Vector2?

### DISCUSSION THREAD

**Discussion by minxuanz** [dotnet/runtime#99396] [Q&A] [answered]
code is `public static Vector2 CreateFromScalar() => new Vector2(1.0f);`
it generates `call     [System.Numerics.Vector2:.ctor(float):this]` when `COMPlus_EnableHWIntrinsic=0 `
and DOTNET_JitDump=vector2:.ctor is  
```
****** START compiling System.Numerics.Vector2:.ctor(float):this
IL to import:
 IL_0000  02                ldarg.0
 IL_0001  03                ldarg.1
 IL_0002  03                ldarg.1
 IL_0003  28 84 2a 00 06    call         0x6002A84
 IL_0008  00                nop
 IL_0009  00                nop
 IL_000a  2a                ret
```
Where was this imported from?
but    https://github.com/dotnet/runtime/blob/4e86b1c63d9c41c6bfb6f42710be907199ce2671/src/libraries/System.Numerics.Vectors/ref/System.Numerics.Vectors.cs#L404-L409

**Comment by huoyaoyuan** [comment, marked as answer]
The managed source is now in `System.Private.CoreLib`:

https://github.com/dotnet/runtime/blob/4e86b1c63d9c41c6bfb6f42710be907199ce2671/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector2.cs#L30-L33

At runtime the body is not used and JIT specially expands it:

https://github.com/dotnet/runtime/blob/4e86b1c63d9c41c6bfb6f42710be907199ce2671/src/coreclr/jit/simdashwintrinsic.cpp#L1761-L1805
